flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesStateHandleStoreTest.java

```diff
@@ -272,6 +327,101 @@ public void testReplaceWithKeyNotExist() throws Exception {
+                            assertThat(newState.isDiscarded(), is(true));
+                        });
+            }
+        };
+    }
+
+    @Test
+    public void testReplaceWithDeletingKeyWithFailingDiscard() throws Exception {
+        new Context() {
+            {
+                runTest(
+                        () -> {
+                            leaderCallbackGrantLeadership();
+
+                            final Exception discardException = new Exception("Unable to discard.");
+                            final TestingLongStateHandleHelper.LongStateHandle oldState =
+                                    addDeletingEntry(
+                                            getLeaderConfigMap(),
+                                            key,
+                                            new TestingLongStateHandleHelper.LongStateHandle(
+                                                    1337,
+                                                    discardIdx -> {
+                                                        throw discardException;
+                                                    }));
+
+                            final KubernetesStateHandleStore<
+                                            TestingLongStateHandleHelper.LongStateHandle>
+                                    store =
+                                            new KubernetesStateHandleStore<>(
+                                                    flinkKubeClient,
+                                                    LEADER_CONFIGMAP_NAME,
+                                                    longStateStorage,
+                                                    filter,
+                                                    LOCK_IDENTITY);
+                            final TestingLongStateHandleHelper.LongStateHandle newState =
+                                    new TestingLongStateHandleHelper.LongStateHandle(23456L);
+
+                            assertThat(store.exists(key), is(StringResourceVersion.notExisting()));
+                            final StateHandleStore.NotExistException exception =
+                                    assertThrows(
+                                            StateHandleStore.NotExistException.class,
+                                            () ->
+                                                    store.replace(
+                                                            key,
+                                                            StringResourceVersion.notExisting(),
+                                                            newState));
+                            assertThat(exception.getSuppressed().length, is(1));
+                            assertThat(
+                                    exception.getSuppressed()[0],
+                                    FlinkMatchers.containsCause(discardException));
+
+                            // Only the new handle should have been stored and discarded.
```

Comment:
```suggestion
                            // Only the new handle should have been discarded.
```
The comment doesn't make sense because also the old state is stored, still

Reply:
👍